src/field.rs

```diff
@@ -798,13 +833,226 @@ pub(crate) fn decode_fieldvec<F: FieldElement>(
     Ok(vec)
 }
 
+#[cfg(test)]
+pub(crate) mod test_utils {
+    use super::{FieldElement, FieldElementWithInteger};
+    use crate::{codec::CodecError, field::FieldError, prng::Prng};
+    use assert_matches::assert_matches;
+    use std::{
+        collections::hash_map::DefaultHasher,
+        convert::{TryFrom, TryInto},
+        fmt::Debug,
+        hash::{Hash, Hasher},
+        io::Cursor,
+        ops::{Add, BitAnd, Div, Shl, Shr, Sub},
+    };
+
+    /// A test-only copy of `FieldElementWithInteger`.
+    ///
+    /// This trait is only used in tests, and it is implemented on some fields that do not have
+    /// `FieldElementWithInteger` implementations. This separate trait is used in order to avoid
+    /// affecting trait resolution with conditional compilation. Additionally, this trait only
```

Comment:
My initial reasoning was to avoid interactions between conditional compilation and trait resolution. (i.e., code that compiles successfully in some settings, but is too ambiguous in others, for non-intuitive reasons) The second, and more important, reason is that the existing code includes a `Copy` bound on the `Integer` associated type. We can meet this with `u32`, `u64`, and `u128`, but not `num_bigint::BigUint`. This test-only trait only requires that the integer provide `Clone`.